classes/ThankYous/ThankYouAcl.php

```diff
@@ -0,0 +1,132 @@
+	 */
+	public function __construct(AdminPanel $admin_panel, UserExtranetService $user_extranet_service)
+	{
+		$this->admin_panel   = $admin_panel;
+		$this->user_extranet = $user_extranet_service;
+	}
+
+	/**
+	 * Determines whether a Security Context can Delete a Thank You.
+	 *
+	 * @param ThankYou        $thank_you
+	 * @param SecurityContext $context
+	 * @return bool
+	 */
+	public function CanDeleteThankYou(ThankYou $thank_you, SecurityContext $context): bool
+	{
+		return $thank_you->GetAuthor()->GetId() === $context->GetUser()->GetId() || $this->IsAdmin($context);
+	}
+
+	/**
+	 * Determines whether a Security Context can Edit a Thank You.
+	 *
+	 * @param ThankYou        $thank_you
+	 * @param SecurityContext $context
+	 * @return bool
+	 */
+	public function CanEditThankYou(ThankYou $thank_you, SecurityContext $context): bool
+	{
+		return $thank_you->GetAuthor()->GetId() === $context->GetUser()->GetId() || $this->IsAdmin($context);
+	}
+
+	/**
+	 * Determines whether a Security Context has access to the Thank You Admin.
+	 *
+	 * @param SecurityContext $context
+	 * @return bool
+	 */
+	public function IsAdmin(SecurityContext $context): bool
+	{
+		return $this->admin_panel->IsAccessible($context);
+	}
+
+	/**
+	 * Determines whether a Security Context can view a Thankable's Name.
+	 *
+	 * @param SecurityContext $context
+	 * @param Thankable       $thankable
+	 * @return bool
+	 */
+	public function CanSeeThankableName(SecurityContext $context, Thankable $thankable): bool
+	{
+		$thankable_extranet_id = $thankable->GetExtranetId();
+
+		return !isset($thankable_extranet_id) || $this->IsExtranetVisible($thankable_extranet_id, $context->GetExtranetAreaId());
+	}
+
+	/**
+	 * Determines whether a Security Context can view a Thank You Author's Name.
+	 * If the Author's Extranet is not set, `false` is returned.
+	 *
+	 * @param SecurityContext $context
+	 * @param ThankYou        $thank_you
+	 * @return bool
+	 */
+	public function CanSeeThankYouAuthor(SecurityContext $context, ThankYou $thank_you): bool
+	{
+		return $this->CanSeeUser($context, $thank_you->GetAuthor());
+	}
+
+	/**
+	 * Determines whether a Security Context can view a User's details.
+	 * If the User's Extranet is not set, `false` is returned.
+	 *
+	 * @param SecurityContext $context
+	 * @param User            $user
+	 * @return bool
+	 */
+	public function CanSeeUser(SecurityContext $context, User $user): bool
+	{
+		$user_extranet_id = $user->GetExAreaId();
+
+		if (!isset($user_extranet_id))
+		{
+			return false;
+		}
+
+		return $this->IsExtranetVisible($user_extranet_id, $context->GetExtranetAreaId());
+	}
+
+	/**
+	 * Determines whether an Extranet Area is visible. If the second parameter is provided, this will be relative to
+	 * that Extranet.
+	 *
+	 * @param int      $target_extranet_id
+	 * @param int|null $viewers_extranet_id
+	 * @return bool
+	 */
+	public function IsExtranetVisible(int $target_extranet_id, ?int $viewers_extranet_id = null): bool
```

Comment:
I don't see why this couldn't just take a `SecurityContext` instead

Reply:
I've made the change so that the method takes a SecurityContext as the first argument, and the target extranet as the second, which is more standard to how ACLs work in the application generally.

I don't fully agree with this change, though I think that is more to do with the belief that this function offers a level of utility that should already be in core (probably hidden behind an API layer which does take a SecurityContext) but isn't. It would be nice to see this level of control added in future refactoring, and this method deprecated as a result.

Fixed in 06eb85e518264f5e3e577223f9cdc0734ce3fbd5

Reply:
Indeed, and in 8.9 now it's available from `PeopleAcl::CanViewExtranet($context, $extranet)`, where `$extranet` can be an Extranet ID or Extranet entity.

Still, this change looks tidy to me, now. Feel free to swap it for using `PeopleAcl` if you fancy, now that People API v2 has indeed been merged.